py-polars/polars/_utils/construction/dataframe.py

```diff
@@ -1020,6 +1020,11 @@ def pandas_to_pydf(
 ) -> PyDataFrame:
     """Construct a PyDataFrame from a pandas DataFrame."""
     convert_index = include_index and not _pandas_has_default_index(data)
+    if convert_index and set(data.index.names).intersection(data.columns):
```

Comment:
Polars converts non-string column names to string

I think a more generic solution is needed, to handle https://github.com/pola-rs/polars/issues/16025 too